Compilation error on MacOS

Hi 🙋‍♀️ 

Running `cmake .` in the freshly copied repo works fine. Compiling however does not:
Executing `make` results in this 
```
[ 33%] Building C object CMakeFiles/TicTacToe.dir/main.c.o
/Users/teufelchen/Programming/TicTacToe/main.c:20:10: fatal error: 'SDL2/SDL.h' file not found
#include <SDL2/SDL.h>
         ^~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/TicTacToe.dir/main.c.o] Error 1
make[1]: *** [CMakeFiles/TicTacToe.dir/all] Error 2
make: *** [all] Error 2
```

A work around is to run gcc manually:
`gcc main.c include/graphics.c -I/usr/local/include -L/usr/local/lib -lSDL2 -lSDL2_TTF `

However upon executing the program:
```
Welcome to Tic Tac Toe! Copyright (C) 2021
Couldn't load small font! Reason: Couldn't open ../resources/Serif.ttf
Couldn't load small font! Reason: Couldn't open ../resources/Serif.ttf
[1]    74802 segmentation fault  ./a.out
```

Which clearly sucks. 

So we got multiple issues here:

1. The readme does not tell you how to compile and run the program(it should also listen the dependency)
2. `*.ttf` not included in the repo
3. Missing error handling when the Fonts could not be loaded
4. Wrong error message

I would kindly ask you to confirm these issues(it might still be user-error on my side!) and then open separate Github issues for them. 

Thanks in advance!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,10 +8,26 @@ set(SOURCE_FILES ${SOURCE_FILES} include/graphics.c)
 
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
 
+if(${CMAKE_HOST_APPLE})
+    set(SDL2_TTF_INCLUDE_DIRS "/usr/local/Cellar/sdl2_ttf/2.0.15/include/SDL2")
+    set(SDL2_TTF_LIBRARIES "-L/usr/local/Cellar/sdl2_ttf/2.0.15/lib" "-lSDL2_TTF")
+else()
 INCLUDE(FindPkgConfig)
 
 pkg_check_modules(SDL2 REQUIRED sdl2)
 pkg_check_modules(SDL2_TTF REQUIRED SDL2_ttf)
+endif()
+
+IF (${SDL2_FOUND})
+    Message( STATUS "SDL2_INCLUDE_DIR:" ${SDL2_INCLUDE_DIRS})
+    Message( STATUS "SDL2_TTF_INCLUDE_DIR:" ${SDL2_TTF_INCLUDE_DIRS})
+    Message( STATUS "SDL2_LIBRARY: " ${SDL2_LIBRARIES})
+    Message( STATUS "SDL2_TTF_LIBRARY: " ${SDL2_TTF_LIBRARIES})
+ELSE()
+    Message( STATUS "SDL2_FOUND: " ${SDL2_FOUND})
+    Message( FATAL_ERROR "SDL2 NOT FOUND" )
+ENDIF()
+
 
 include_directories(${SDL2_INCLUDE_DIRS}
         ${SDL2_TTF_INCLUDE_DIRS})

--- a/include/common.h
+++ b/include/common.h
@@ -5,8 +5,14 @@
 #ifndef TICTACTOE_COMMON_H
 #define TICTACTOE_COMMON_H
 
-int playerScore;
-int aiScore;
-int draws;
+#ifdef MAIN_INCLUDE
+#define EXTERN
+#else
+#define EXTERN extern
+#endif
+
+EXTERN int playerScore;
+EXTERN int aiScore;
+EXTERN int draws;
 
 #endif //TICTACTOE_COMMON_H

--- a/include/graphics.c
+++ b/include/graphics.c
@@ -15,6 +15,7 @@ You should have received a copy of the GNU General Public License
 along with TicTacToe.  If not, see <https://www.gnu.org/licenses/>.
 */
 
+#define MAIN_INCLUDE
 #include "graphics.h"
 #include "common.h"
 #if !defined(NXDK)
@@ -85,18 +86,27 @@ void InitStdText() {
     smallFont = TTF_OpenFont(RESOURCEDIR"Serif.ttf", 30);
     if (!smallFont) {
         printf("Couldn't load small font! Reason: %s\n", SDL_GetError());
+        TTF_Quit();
+        SDL_Quit();
         return;
     }
 
     bigFont = TTF_OpenFont(RESOURCEDIR"Serif.ttf", 58);
     if (!bigFont) {
         printf("Couldn't load big font! Reason: %s\n", SDL_GetError());
+        TTF_CloseFont(smallFont);
+        TTF_Quit();
+        SDL_Quit();
         return;
     }
 
     ultraSmallFont = TTF_OpenFont(RESOURCEDIR"Serif.ttf", 25);
     if (!bigFont) {
         printf("Couldn't load ultra small font! Reason: %s\n", SDL_GetError());
+        TTF_CloseFont(bigFont);
+        TTF_CloseFont(smallFont);
+        TTF_Quit();
+        SDL_Quit();
         return;
     }
 

--- a/include/graphics.h
+++ b/include/graphics.h
@@ -26,18 +26,24 @@ along with TicTacToe.  If not, see <https://www.gnu.org/licenses/>.
 #include <SDL_ttf.h>
 #endif
 
-int windowWidth;
-int windowHeight;
-TTF_Font *smallFont;
-TTF_Font *bigFont;
-TTF_Font *ultraSmallFont;
-SDL_Surface *bgSurface;
-SDL_Window *window;
-SDL_Surface *windowSurface;
-Uint32 Rmask;
-Uint32 Gmask;
-Uint32 Bmask;
-Uint32 Amask;
+#ifdef MAIN_INCLUDE
+#define EXTERN
+#else
+#define EXTERN extern
+#endif
+
+EXTERN int windowWidth;
+EXTERN int windowHeight;
+EXTERN TTF_Font *smallFont;
+EXTERN TTF_Font *bigFont;
+EXTERN TTF_Font *ultraSmallFont;
+EXTERN SDL_Surface *bgSurface;
+EXTERN SDL_Window *window;
+EXTERN SDL_Surface *windowSurface;
+EXTERN Uint32 Rmask;
+EXTERN Uint32 Gmask;
+EXTERN Uint32 Bmask;
+EXTERN Uint32 Amask;
 
 
 void screenInit();

